### PR TITLE
Lazy loading of finished jobs

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java [hunk 1]
```diff
@@ -1245,12 +1241,29 @@ public UserData getCurrentUserData() throws NotConnectedException {
         return PASchedulerProperties.getPropertiesAsHashMap();
     }
 
+    synchronized ClientJobState getClientJobState(JobId jobId) {
+        if (!jobsMap.containsKey(jobId)) {
+            List<InternalJob> internalJobs = dbManager.loadInternalJob(Arrays.asList(jobId.longValue()));
+            if (!internalJobs.isEmpty()) {
+                InternalJob internalJob = internalJobs.get(0);
+                ClientJobState clientJobState = new ClientJobState(internalJob);
+                jobsMap.put(jobId, clientJobState);
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: Beware that by doing jobsMap.put(), the job will be kept in memory without housekeeping

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java [hunk 2]
```diff
@@ -130,7 +130,7 @@ public RecoveredSchedulerState recover(long loadJobPeriod, RMProxy rmProxy) {
             }
         }
 
-        finishedJobs.addAll(dbManager.loadFinishedJobs(false, loadJobPeriod));
+        //        finishedJobs.addAll(dbManager.loadFinishedJobs(false, loadJobPeriod));
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java]: We shouldn't do this but use the property instead

Comment (Mykhailenko) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java]: Ok

**Comment by Mykhailenko** [comment]
@sophiesongge you are right, i will change to make it true LRU

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java [hunk 3]
```diff
@@ -1245,12 +1253,33 @@ public UserData getCurrentUserData() throws NotConnectedException {
...
+    synchronized ClientJobState getClientJobState(JobId jobId) {
+        if (!jobsMap.containsKey(jobId)) {
+            if (!finishedJobsLRUCache.containsKey(jobId)) {
+                List<InternalJob> internalJobs = dbManager.loadInternalJob(jobId.longValue());
+                if (!internalJobs.isEmpty()) {
+                    InternalJob internalJob = internalJobs.get(0);
+                    ClientJobState clientJobState = new ClientJobState(internalJob);
+                    finishedJobsLRUCache.put(jobId, clientJobState);
+                }
+            }
```
Comment (sophiesongge) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: If the cache already contains the key, we need to remove it first, then add it to the head of the linkedhashmap

**Pull request merged**